registrations/permissions.py

```diff
@@ -48,7 +48,10 @@ def has_object_permission(
             if obj.data_source != user_data_source:
                 return False
 
-        return obj.can_be_edited_by(request.user)
+        # check_object_permissions is called from RegistrationViewSet perform_create and
+        # perform_update methods with event value, so use Registration.can_be_edited_by
+        # instead of obj.can_be_edited_by to use correct can_be_edited_by method
+        return Registration.can_be_edited_by(obj, request.user)
```

Comment:
Just a note that the signature of this method is wrong, since `obj` is not necessarily of type `Registration`.

Could the `perform_update` actually be deleted since, since by default `get_object` will call `check_object_permissions` with the `Registration` instance. Or is it possible to update the event field on a registration?

Maybe for the `perform_create` instead of calling `check_object_permissions` we could either try to improve the POST check in `CanAccessRegistration.has_permission` or just do the required check in `perform_create` or in the serializer's `create` method.

Reply:
also some test(s) for the case would be awesome to validate the solution.

Reply:
Great comment! 

I did following changed to get rid of the hack:
- Check event data source for api key user in `RegistrationSerializer` `create` method
- Make `event` read_only field in `RegistrationSerializer` `__init__` method. In that case there is no need for the permission check of the event object.